### PR TITLE
Now with QueueFull and QueueEmpty

### PR DESCRIPTION
**Pull request by raysalemi**
This has the new cocotb.queue.QueueFull and cocotb.queue.QueueEmpty exceptions.

### CODE REVIEW

File: cocotb/queue.py [hunk 1]
```diff
@@ -4,11 +4,20 @@
 from typing import Generic, TypeVar
 import collections
 import heapq
-from asyncio import QueueEmpty, QueueFull
 
+import asyncio.queues
```
Comment (eric-wieser) [inline cocotb/queue.py:8]:
```suggestion
import asyncio.queues

```

File: cocotb/queue.py [hunk 2]
```diff
@@ -4,11 +4,20 @@
 from typing import Generic, TypeVar
 import collections
 import heapq
-from asyncio import QueueEmpty, QueueFull
```
Comment (eric-wieser) [inline cocotb/queue.py:7]:
```suggestion
```

**Review by eric-wieser — COMMENTED** [review]
Asyncio should be grouped with the other built-in imports

**Review by ktbarrett — APPROVED** [review]

File: cocotb/queue.py [hunk 3]
```diff
@@ -4,11 +4,20 @@
...
 import cocotb
 from cocotb.triggers import Event, _pointer_str
 
+
+class QueueFull(asyncio.queues.QueueFull):
+    """Raised when the Queue.put_nowait() method is called on a full Queue."""
+
+
+class QueueEmpty(asyncio.queues.QueueEmpty):
+    """Raised when the Queue.get_nowait() or Queue.peek_nowait() method is called on a empty Queue."""
```
Comment (marlonjames) [inline cocotb/queue.py:18]:
```suggestion
    """Raised when the Queue.get_nowait() method is called on a empty Queue."""
```

**Review by marlonjames — APPROVED** [review]

**Comment by ktbarrett** [comment]
[It appears in the docs](https://cocotb--2631.org.readthedocs.build/en/2631/library_reference.html#module-cocotb.queue). I don't think it's worth a newsfragment TBH. So merging.

**Pull request merged**